File: src/1480-Running-Sum-of-1d-Array/running_sum_of_1d_array.cpp
#include <iostream>
#include <vector>
// Definition for a binary tree node.
class Solution {
public:
    std::vector<int> runningSum(std::vector<int>& nums) {
        std::vector<int> output;
        for(int i = 0; i< nums.size(); i++){
            if(i == 0){
                output.push_back(nums[i]);
            }
            else{
                output.push_back(nums[i] + output[i - 1]);
            }
        }
      return output;
    }
};
      
int main() {
  std::vector<int> nums {1,2,3,4};
  Solution solution;
  std::vector<int> result = solution.runningSum(nums);
  
  for (int i : result) {
    std::cout << i << " ";
  }
  
  return 0;
}

File: src/1480-Running-Sum-of-1d-Array/go/running_sum_of_1d_array.go
package main

import "fmt"

func runningSum(nums []int) []int {
	res := make([]int, 0, len(nums))
	for i := 0; i < len(nums); i++ {
		if i == 0 {
			res = append(res, nums[i])
		} else {
			new_num := nums[i] + res[i-1]
			res = append(res, new_num)
		}
	}
	return res
}

func main() {
	nums := []int{1, 2, 3, 4}
	res := runningSum(nums)
	fmt.Print(res)
}
